client/src/pages/Userprofile/UserWeatherSection.js

```diff
@@ -0,0 +1,147 @@
+import React, { useEffect, useState } from 'react';
+import { useGeolocation } from '../UserLocation/useGeolocation';
+import { styled } from '@mui/material';
+
+const WeatherSection = styled('section')`
+  margin: 48px;
+  width: 300px;
+  height: 200px;
+  border-radius: 6px;
+  background-color: rgb(127, 210, 244);
+  color: white;
+  padding: 20px;
+  display: flex;
+  flex-direction: column;
+`;
+
+const TemperatureArticle = styled('article')`
+  font-size: 32px;
+  border-right: 1px solid white;
+  padding-right: 16px;
+  margin-right: 16px;
+`;
+
+const DateArticle = styled('article')`
+  display: flex;
+  align-items: center;
+`;
+
+const MoreInfoContainer = styled('div')`
+  display: flex;
+  justify-content: space-between;
+  margin-top: auto;
+  margin-bottom: 0;
+  align-self: bottom;
+`;
+
+const ValueParagraph = styled('p')`
+  font-size: 20px;
+`;
+
+const ONE_HOUR = 1000 * 60 * 60;
+const WEATHER_CACHE = 'WEATHER_CACHE';
+
+const directions = [
+  'N',
+  'NNE',
+  'NE',
+  'ENE',
+  'E',
+  'ESE',
+  'SE',
+  'SSE',
+  'S',
+  'SSW',
+  'SW',
+  'WSW',
+  'W',
+  'WNW',
+  'NW',
+  'NNW',
+];
+
+export default function UserWeatherSection() {
+  const currentDate = new Date();
+  const month = currentDate.toLocaleString('default', { month: 'long' });
+  const weekday = currentDate.toLocaleString('default', { weekday: 'long' });
+  const day = currentDate.getDate();
+
+  const { data } = useGeolocation({ watching: false });
+
+  // Pleae, add your API_KEY
+  const API_KEY = 'write_the_key_here';
```

Comment:
Is this meant to be updated?

Reply:
Yes, as I wrote I am using my API_KEY, I think it is better the project owners to create API_KEY for the project and use it.

Reply:
Yup, I see that zoobot responded above
> Does open weather have an allow list of urls for the key? We usually add only our dev.waterthetrees.com and waterthetrees.com to the allow list once we are done testing. You can add the key here once it's locked down to only our urls: https://github.com/waterthetrees/wtt_front/blob/main/client/src/util/config.js

Did you see this response? It looks like step 1: configure the account tied to the API key to only allow a whitelist of URLs to use it and step 2: commit the key to `config.js`. Can you try this?

If you can't create a whitelist, then we need to store the API key within an environment variable.